lib/versioning/Version.js

```diff
@@ -0,0 +1,168 @@
+'use strict'; // eslint-disable-line strict
+
+/**
+ * Class for manipulating an object version.
+ * The format of a version: { isNull, isDeleteMarker, versionId, otherInfo }
+ */
+class Version {
+    /**
+     * Create a new version instantiation from its data object.
+     * @param {object} version - the data object to instantiate
+     * @param {boolean} version.isNull - is a null version
+     * @param {boolean} version.isDeleteMarker - is a delete marker
+     * @param {string} version.versionId - the version id
+     * @constructor
+     */
+    constructor(version) {
+        this.version = version || {};
+    }
+
+    /**
+     * Parse the version information from a string.
+     *
+     * @param {string} value - the string to parse
+     * @return {Version} - the version deserialized from the input string
+     */
+    static from(value) {
+        return new Version(value ? JSON.parse(value) : undefined);
+    }
+
+    /**
+     * [MetaData Internal] Check if a version is a place holder for deletion.
+     *
+     * @param {string} value - version to check
+     * @return {boolean} - whether this is a PHD version
+     */
+    static isPHD(value) {
```

Comment:
IMO, internal functions need to be self-documented as much as possible, especially because internal tends to have more intricate complexity, if not for the API user since they're internal, but for a programmer who may have to dig into it later. It's good that the meaning of PHD is explained somewhere, but I think it would be even better to have a more descriptive name for the function name itself as well.